#2561 Comments refinement

ccanos opened the pull request:
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

hero101 commented on src/domain/aspect/views/AspectDashboardView.tsx:
```diff
@@ -171,7 +168,6 @@ const AspectDashboardView: FC<AspectDashboardViewProps> = props => {
                   placeholder={t('pages.aspect.dashboard.comment.placeholder')}
                   onPostComment={onPostComment}
                   maxLength={MID_TEXT_LENGTH}
-                  userAvatarUri={user?.user?.profile?.avatar?.uri}
```
why was this required?

ccanos replied:
Before the restyling the `PostComment` component was just a Markdown without the user avatar in it. After Aleks's changes it has the avatar on the left of the input field.
This is some code that I let Aleks merge to refactor it later because Neil told us to close Aleks'es task and let me continue on this other one. 
You can see his changes here: https://github.com/alkem-io/client-web/pull/2549/files
He added the userContext to the AspectDashboardView  only to pass the userAvatar to the PostComment form field. 
I didn't like that very much so I have moved that code to the PostComment.

After my last changes PostComment is renamed to PostMessageToCommentsForm

hero101 commented on src/domain/communication/useAuthorsDetails.ts:
```diff
@@ -3,26 +3,89 @@ import { Author } from '../../models/discussion/author';
+  const { opportunityId } = useOpportunity();
+
+  const resourceId = opportunityId || challengeId || hubId || '';
+  const reourceType: ResourceTypeName = opportunityId ? 'opportunity' : challengeId ? 'challenge' : 'hub';
```
typo in var name

ccanos replied:
fixed, thank you

hero101 commented on src/models/constants/translation.constants.ts:
```diff
@@ -1,5 +1,8 @@
+export const HOST_TRANSLATION_KEY = 'common.host';
 
 export type RoleNameKey = typeof OWNER_TRANSLATION_KEY | typeof ADMIN_TRANSLATION_KEY | typeof MEMBER_TRANSLATION_KEY;
+export type AuthorRoleNameKey = typeof LEAD_TRANSLATION_KEY | typeof HOST_TRANSLATION_KEY;
```
this does not give enough context to where it's used. `[Comment/Message]Author` gives more context imo.

ccanos replied:
`MessageAuthorRoleNameKey` I like it more too

hero101 reviewed (changes requested):
My main issues are things missed on the last PR about comments.
In general I think that the "comments" code must stop living under Discussions.
For example:
- the comment component is not a "Discussions comment"
- the author avatar component should not be specially built for discussions (or live under discussions)

ccanos commented:
@hero101 I have refactored and moved all Comments stuff to `domain/shared/components/Comments` 
I like that more too. Changed a bit some names to make proper distinction between a Comments object, and a Message
